service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/GeneratedOrderController.java

```diff
@@ -142,15 +146,18 @@ public AboutToStartOrSubmitCallbackResponse handleAboutToSubmit(
     }
 
     @PostMapping("/submitted")
-    public void handleSubmittedEvent(@RequestHeader(value = "authorization") String authorization,
-                                     @RequestHeader(value = "user-id") String userId,
-                                     @RequestBody CallbackRequest callbackRequest) {
+    public void handleSubmittedEvent(@RequestBody CallbackRequest callbackRequest) {
         CaseData caseData = mapper.convertValue(callbackRequest.getCaseDetails().getData(), CaseData.class);
         String mostRecentUploadedDocumentUrl = service.getMostRecentUploadedOrderDocumentUrl(
             caseData.getOrderCollection());
 
-        applicationEventPublisher.publishEvent(new GeneratedOrderEvent(callbackRequest, authorization, userId,
-            concatGatewayConfigurationUrlAndMostRecentUploadedOrderDocumentPath(mostRecentUploadedDocumentUrl)));
+        byte[] documentContents = documentDownloadService.downloadDocument(requestData.authorisation(),
```

Comment:
so my understanding of requestData, you can just declare it in DocumentDownloadService. Using it like this doesn't actually provide any benefits over declaring it in the method signature. I would recommend moving and only sending mostRecentUploadedDocumentUrl

Reply:
I agree, happy to add RequestData to DocumentDownloadService although it might inflate the PR a little? thoughts?